File: src/insert_sort.cc

#include "../include/insert_sort.h"
#include <iostream>
using namespace std;
template<typename T>
void insert_sort(T *arr, int n){
    if( n<=1){
        return ;
    }
    //T key = arr[0];
    for (int i=1; i<n; i++){
        for(int j=i-1;j>=0;j--){
            T tmp;
            //cout<<arr[j]<<endl;
            if(arr[i]<arr[j] && arr[i]>=arr[j-1]){
                tmp = arr[i];
                for(int z=i-1;z>=j;z--){
                    arr[z+1] = arr[z];
                }
                arr[j] = tmp;
            }
            
           // cout<<i<<"   "<<arr[j]<<endl;
        }
        //cout<<arr[i]<<endl;
    }
}

File: src/insert_sort.js
var arr=function(a,n){
	for(let i=0;i<n-1;i++){
		var min = i;
		var k=a[min];
		for(let j=i+1;j<n;j++){
			if(a[j]<a[min]){
				min=j;
			}
		}

		if(min!=i){
			var tmp=a[min];a[min]=a[i];a[i]=tmp;
		}
	}
	return a;
}

var arr1=[2,4,1,6,3,5];
console.log(arr(arr1,6));